capture_service/dive_client_cli.cc

```diff
@@ -22,421 +22,389 @@ limitations under the License.
+    auto* device = ctx.mgr.GetDevice();
+    if (device == nullptr)
     {
-        std::cout << "Missing required options." << std::endl;
-        PrintUsage();
-        return false;
+        return absl::FailedPreconditionError(
+        "No device selected. Did you provide --device serial?");
     }
-    auto dev_ret = mgr.SelectDevice(serial);
+    device->EnableGfxr(enable_gfxr);
 
-    if (!dev_ret.ok())
-    {
-        std::cout << "Failed to select device " << dev_ret.status().message() << std::endl;
-        return false;
-    }
-    auto dev = *dev_ret;
-    dev->EnableGfxr(is_gfxr_capture);
-    auto        ret = dev->SetupDevice();
-    std::string gfxr_capture_arg = "gfxr_capture";
-    if (!ret.ok())
-    {
-        std::cout << "Failed to setup device, error: " << ret.message() << std::endl;
-        return false;
-    }
+    absl::Status ret;
 
-    if (app_type == "vulkan_openxr")
-    {
-        ret = dev->SetupApp(package,
-                            Dive::ApplicationType::OPENXR_APK,
-                            command_args,
-                            device_architecture,
-                            gfxr_capture_directory);
-    }
-    else if (app_type == "vulkan_non_openxr")
-    {
-        ret = dev->SetupApp(package,
-                            Dive::ApplicationType::VULKAN_APK,
-                            command_args,
-                            device_architecture,
-                            gfxr_capture_directory);
-    }
-    else if (app_type == "vulkan_cli_non_openxr")
+    switch (ctx.options.app_type)
     {
-        ret = dev->SetupApp(command,
-                            command_args,
-                            Dive::ApplicationType::VULKAN_CLI,
-                            device_architecture,
-                            gfxr_capture_directory);
+    case AppType::kVulkan_OpenXR:
+        ret = device->SetupApp(ctx.options.package,
+                               Dive::ApplicationType::OPENXR_APK,
+                               ctx.options.vulkan_command_args,
+                               ctx.options.device_architecture,
+                               ctx.options.gfxr_capture_file_dir);
+        break;
+    case AppType::kVulkan_Non_OpenXR:
+        ret = device->SetupApp(ctx.options.package,
+                               Dive::ApplicationType::VULKAN_APK,
+                               ctx.options.vulkan_command_args,
+                               ctx.options.device_architecture,
+                               ctx.options.gfxr_capture_file_dir);
+        break;
+    case AppType::kVulkanCLI_Non_OpenXR:
+        ret = device->SetupApp(ctx.options.vulkan_command,
+                               ctx.options.vulkan_command_args,
+                               Dive::ApplicationType::VULKAN_CLI,
+                               ctx.options.device_architecture,
+                               ctx.options.gfxr_capture_file_dir);
+        break;
+    case AppType::kGLES_OpenXR:
```

Comment:
https://github.com/google/dive/pull/561 is adding a new type that you probably need to check